android/src/main/java/com/reactnativedetector/ScreenshotDetectionDelegate.kt

```diff
@@ -5,82 +5,109 @@ import android.content.Context
-    }
+  fun stopScreenshotDetection() {
+     val contentObserver: ContentObserver = object : ContentObserver(Handler(Looper.getMainLooper())) {
+      override fun onChange(selfChange: Boolean, uri: Uri?) {
+
+        super.onChange(selfChange, uri)
+        if (!isReadExternalStoragePermissionGranted()) {
+          onScreenCapturedWithDeniedPermission()
+        }
 
-    private fun onScreenCapturedWithDeniedPermission() {
-        listener.onScreenCapturedWithDeniedPermission()
+        if (uri != null) {
+          val path = getFilePathFromContentResolver(context, uri)
+          if (path != null && isScreenshotPath(path)) {
+            previousPath = path.toLowerCase().substring(path.toLowerCase().lastIndexOf("screenshot"))
+            onScreenCaptured(path!!)
+          }
+        }
+      }
     }
 
-    private fun isScreenshotPath(path: String?): Boolean {
-        return path != null && path.toLowerCase().contains("screenshots") && previousPath != path
+    context.getContentResolver().unregisterContentObserver(contentObserver)
+    isListening = false
+  }
+
+  private fun onScreenCaptured(path: String) {
+    listener.onScreenCaptured(path)
+  }
+
+  private fun onScreenCapturedWithDeniedPermission() {
+    listener.onScreenCapturedWithDeniedPermission()
+  }
+
+  private fun isScreenshotPath(path: String?): Boolean {
+        return path != null && path.toLowerCase().contains("screenshots") && (previousPath == "" || !path.toLowerCase().contains(previousPath))
     }
 
-    private fun getFilePathFromContentResolver(context: Context, uri: Uri): String? {
-        try {
+  private fun getFilePathFromContentResolver(context: Context, uri: Uri): String? {
+    try {
 
-            val cursor = context.contentResolver.query(uri, arrayOf(MediaStore.Images.Media.DISPLAY_NAME, MediaStore.Images.Media.DATA), null, null, null)
-            if (cursor != null && cursor.moveToFirst()) {
-                val path = cursor.getString(cursor.getColumnIndex(MediaStore.Images.Media.DATA))
-                cursor.close()
-                return path
-            }
-        } catch (e: Exception) {
+      val cursor = context.contentResolver.query(uri, arrayOf(MediaStore.Images.Media.DISPLAY_NAME, MediaStore.Images.Media.DATA), null, null, null)
+      if (cursor != null && cursor.moveToFirst()) {
+        val path = cursor.getString(cursor.getColumnIndex(MediaStore.Images.Media.DATA))
+        cursor.close()
+        return path
+      }
+    } catch (e: Exception) {
 
-        }
-        return null
     }
+    return null
+  }
+
+  private fun isReadExternalStoragePermissionGranted(): Boolean {
+    val isAndroid13OrAbove = Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU
 
-    private fun isReadExternalStoragePermissionGranted(): Boolean {
-        return ContextCompat.checkSelfPermission(context, Manifest.permission.READ_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED
+    if (isAndroid13OrAbove) {
```

Comment:
I'm fairly certain that we do not need to check for permissions with using MediaStore and API >= 19 (android 10).  Media Store works under android scope storage model, which allows for a secure way to interact with media under the shared screenshots directory